### PR TITLE
MULTIARCH-3793: Power VS: Add mad and wdc assupported region

### PR DESCRIPTION
**Pull request by mjturek**

### CODE REVIEW

**Comment by miyamotoh** [comment]
Now that this PR is making "s1022" SysType support official, we might want to add/update some validation logic near https://github.com/openshift/installer/blob/e80d50ac8c9ab65833045d648a01eb1a0a3a6d7d/pkg/types/powervs/validation/machinepool_test.go#L134-L139. Otherwise, LGTM.

File: data/data/coreos/rhcos.json [hunk 1]
```diff
@@ -350,6 +350,12 @@
               "bucket": "rhcos-powervs-images-eu-de",
               "url": "https://s3.eu-de.cloud-object-storage.appdomain.cloud/rhcos-powervs-images-eu-de/rhcos-415-92-202310310037-0-ppc64le-powervs.ova.gz"
             },
+            "eu-es": {
```
Comment (hamzy) [inline data/data/coreos/rhcos.json:353]: Aren't these auto generated? Ex: https://github.com/openshift/installer/pull/7654

Comment (mjturek) [reply data/data/coreos/rhcos.json:353]: I believe it is but we don't want to wait for it to get generated to add the region

Comment (hamzy) [reply data/data/coreos/rhcos.json:353]: Pointing to a different region sounds hacky.

Comment (r4f4) [reply data/data/coreos/rhcos.json:353]: There is a PR up to update the json: https://github.com/openshift/installer/pull/7770

Comment (mjturek) [reply data/data/coreos/rhcos.json:353]: awesome! I'll remove it from this PR

**Review by r4f4 — COMMENTED** [review]
/approve

**Pull request merged**